### PR TITLE
Group samples workflow submission

### PR DESCRIPTION
**Pull request by ksierks**
## What does this PR do and why?
Added the ability for a user with at least analyst access to launch pipelines from the group samples table.
Fixes #455.

## Screenshots or screen recordings
image

## How to set up and validate locally
1. Login to irida-next.
2. Create a new project and a new sample within the new project.
3. Go back to project samples page and select the new sample.
4. Verify clicking the pipeline submission button works the same as on the project samples table.

Repeat with a user less than an analyst and verify the pipeline submission button is disabled.

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.

- [x] I have evaluated the [PR acceptance checklist](https://phac-nml.github.io/irida-next/docs/development/development_processes/code_review#acceptance-checklist) for this PR.

### CODE REVIEW

File: app/models/member.rb [hunk 1]
```diff
@@ -121,6 +121,12 @@ def can_update_namespace_with_group_link?(user, object_namespace)
       can_modify?(user, object_namespace)
     end
 
+    def can_submit_workflow?(user, object_namespace)
+      Member::AccessLevel.manageable.include?(
```
Comment (deepsidhu85) [inline app/models/member.rb:125]: The `manageable` access levels are only `maintainer` and `owner`. A user can submit a workflow with a minimum access level of `Analyst`. We will probably want to create another method in the `AccessLevel` class such as `workflow_submittable` or something that maybe sounds a bit better, which we can reuse

File: test/system/workflow_executions/submissions_test.rb [hunk 2]
```diff
@@ -2,37 +2,72 @@
...
+      end
     end
 
-    within 'dialog.dialog--size-xl' do
-      within 'div.sample-sheet' do
-        within 'table' do
-          assert_selector 'tr[data-controller="nextflow--samplesheet"]', count: 2
-          assert_selector 'tr[data-controller="nextflow--samplesheet"]', text: @sample1.puid, count: 1
-          assert_selector 'tr[data-controller="nextflow--samplesheet"]', text: @sample2.puid, count: 1
+    test 'should display a pipeline selection modal for group samples' do
```
Comment (deepsidhu85) [inline test/system/workflow_executions/submissions_test.rb:40]: For this test we should be logging in as an `analyst` to ensure that it works. I see the existing test also logged in as a user with an access level of `owner` which appeared to be missed

**Review by deepsidhu85 — CHANGES_REQUESTED** [review]
This looks great just a few small comments below to address an analyst not being able to launch a workflow

File: test/system/workflow_executions/submissions_test.rb [hunk 3]
```diff
@@ -2,37 +2,72 @@
...
-class SubmissionsTest < ApplicationSystemTestCase
-  setup do
-    @user = users(:john_doe)
-    login_as @user
-    @sample1 = samples(:sample1)
-    @sample2 = samples(:sample2)
-    @project = projects(:project1)
-    @namespace = groups(:group_one)
-  end
+module WorkflowExecutions
```
Comment (joshsadam) [inline test/system/workflow_executions/submissions_test.rb:5]: **praise:** Thanks for fixing this, sorry :)

**Review by joshsadam — APPROVED** [review]
This looks great to me

File: app/models/member.rb (hunk 1 above)
Comment (ksierks) [reply app/models/member.rb]: Is it okay if I do what we did for can_export_data? ed16a029c2c9ee0cc099bf525a0edfc29f386376

File: test/system/workflow_executions/submissions_test.rb (hunk 2 above)
Comment (ksierks) [reply test/system/workflow_executions/submissions_test.rb]: I created a new group & project fixtures to test roles. ed16a029c2c9ee0cc099bf525a0edfc29f386376

File: app/models/member.rb (hunk 1 above)
Comment (deepsidhu85) [reply app/models/member.rb]: Yup that is definitely good! Thanks!

**Review by deepsidhu85 — APPROVED** [review]
Looks good to me!

**Pull request merged**